sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/specialized/BlobOutputStream.java

```diff
@@ -174,7 +182,12 @@ private BlockBlobOutputStream(final BlobAsyncClient client,
                     this.lastError = new IOException(e);
                     return Mono.empty();
                 })
-                .doOnTerminate(() -> complete = true)
+                .doOnTerminate(() -> {
+                    lock.lock();
+                    complete = true;
+                    transferComplete.signal();
```

Comment:
Should I try catch the entire block on the lock and unlock if a generic exception is caught? Or do yall think CI is catching a false positive?